src/Container/ResponseFactory.php

```diff
@@ -21,16 +21,19 @@
  */
 class ResponseFactory
 {
+    /**
+     * @throws Exception\InvalidServiceException
+     */
     public function __invoke(ContainerInterface $container) : ResponseInterface
     {
         if (! class_exists(Response::class)) {
             throw new Exception\InvalidServiceException(sprintf(
                 'The %s service must map to a factory capable of returning an'
-                . ' implementation instance. By default, we assume usage of'
-                . ' zend-diactoros for PSR-7, but it does not appear to be'
-                . ' present on your system. Please install zendframework/zend-diactoros'
-                . ' or provide an alternate factory for the %s service that'
-                . ' can produce an appropriate %s instance.',
+                    . ' implementation instance. By default, we assume usage of'
+                    . ' zend-diactoros for PSR-7, but it does not appear to be'
+                    . ' present on your system. Please install zendframework/zend-diactoros'
+                    . ' or provide an alternate factory for the %s service that'
+                    . ' can produce an appropriate %s instance.',
```

Comment:
Shouldn't these lines start at the same point?

Reply:
And here is another big concern (at least for me). I was thinking about it a lot and I decided to write the sniff that way. I know that PHPStorm is formatting it differently but I think PHPStorm is formatting a lot of things differently/incorrectly (maybe this is configurable), and this is why I didn't take PHPStorm formatting as the correct one, and always have a though how it should be.

So the whole sniff which checks indents is really complicated. I had a lot issues with PHP_CodeSniffer one, so I've decided to write another one which will support everything.

Why I'm adding one more indent here?
For me it's a continuation of the expression - it is a continuation - concatenation in that case.

Usually when we do something like that we also have one more indent:
```php
$var = 'something'
    . ' continue here';
```

or:

```php
return 'hello'
    . ' world';
```

or:

```php
echo $veryLongVariable
    . $anotherVariable;
```

So, basically I think it looks clearer and you easly can see where the parameter is finished and where the next is starting. It's consistent with PSR-2 (I can't see any issues reported by PSR-2 checks), for example:
```php
sprintf(
    'This is'
        . ' very long messsage'
        . ' and it has one more indent on concatenation',
    'param1',
    'param2'
);
```

Let's now think about other operators, like `+`, how it should look like?
```php
method(
    $myVariable // please not this is just example, here could be even more complicated operation
        + ($anotherVariable * $multiplier),
    $secondParam
);
```

What about logical operators, let say:
```php
$this->callSomething(
    $this->checkIfYouCanGoToThePartyTonight($param1, $param2, $param3)
        && $this->havanaClubIsOpenTillMidnight($currentDate),
    $heyItsMe
);
```
Isn't it look clearer than:
```php
$this->callSomething(
    $this->checkIfYouCanGoToThePartyTonight($param1, $param2, $param3)
    && $this->havanaClubIsOpenTillMidnight($currentDate),
    $heyItsMe
);
```

Just another note - I always require to have operator at the beginning of the line, never at the end, so we can't have situation like that - what would be even more confusing:
```php
$this->callSomething(
    $this->checkIfYouCanGoToThePartyTonight($param1, $param2, $param3) && 
    $this->havanaClubIsOpenTillMidnight($currentDate),
    $heyItsMe
);
```

Reply:
Agreed that the operator should not be on the end of the line, but at the start of the next.

I'm torn about these examples:

```php
$this->callSomething(
    $this->checkIfYouCanGoToThePartyTonight($param1, $param2, $param3)
        && $this->havanaClubIsOpenTillMidnight($currentDate),
    $heyItsMe
);
```

and

```php
$this->callSomething(
    $this->checkIfYouCanGoToThePartyTonight($param1, $param2, $param3)
    && $this->havanaClubIsOpenTillMidnight($currentDate),
    $heyItsMe
);
```

The latter is how I've been handling them. You can see a ton of examples particularly with `sprintf()` line continuations:

```php
throw new SomeDomainException(sprintf(
    'This is a really, really, really long line for which I want'
    . ' a continuation to occur. %s and all that',
    $string
));
```

My take has always been that seeing the operator on the subsequent line helps me know that this is a continuation, and thus no additional indentation is necessary. In fact, I tend to think of this as a way to create a visual block, as extra indentation generally implies a _new_ block (that may be due to having played a bit with python in the past, though).

I definitely prefer same level of indentation, but I can also see an argument for the extra indentation, particularly when it comes to argument lists, as it makes it more visually clear when a single argument ends.

I think this is one where we need to poll folks to find out if there's a clear preference.